code/game/gamemodes/gameticker.dm

```diff
@@ -652,20 +661,15 @@ var/datum/controller/gameticker/ticker
 			roles += player.mind.assigned_role
 	return roles
 
-/datum/controller/gameticker/proc/handle_lights() //This is used to turn on lights in occupied departments
-	var/list/discrete_areas = areas.Copy()
-	for(var/mob/living/player in player_list)
-		discrete_areas -= get_department_areas(player)
-
-	for(var/obj/machinery/light_switch/LS in all_machines)
-		if((get_area(LS) in discrete_areas))
-			LS.toggle_switch(0,playsound=FALSE)
-
-	spawn(0)
-		for(var/area/DA in discrete_areas)
-			for(var/obj/item/device/flashlight/lamp/L in DA)
-				sleep(0.1)
-				L.toggle_onoff(0)
+/datum/controller/gameticker/proc/handle_lights(list/areas_to_turn_lights_on)
```

Comment:
A better new name would be "activate_area_lights" or something like that. And I see you made all lamps begin off at roundstart. I see no problem with that, but wanted to bring attention to other reviewers.

Reply:
changed the proc name
i think lamps should be off when created, especially if later they are made craftable it makes more sense theyd be crafted in the off state